src/mono/mono/metadata/appdomain.c

```diff
@@ -613,13 +613,12 @@ mono_domain_owns_vtable_slot (MonoDomain *domain, gpointer vtable_slot)
 	return res;
 }
 
-gboolean
-mono_domain_set_fast (MonoDomain *domain, gboolean force)
+void
+mono_domain_set_fast (MonoDomain *domain)
```

Comment:
there is only one root domain，why not remove this function directly?

Reply:
Setting the domain on the thread is still part of the attach/detach process and hard to completely remove. I looked into replacing the tls variables with a single state variable but it's hard to be sure you've covered every case and such a change would be risky. At any rate, it's definitely out of scope for this PR.